src/lib/bootloader/bls_sections.rb

```diff
@@ -16,27 +16,35 @@ class BlsSections
     # or an empty array
     attr_reader :all
 
-    # @return [String] title of default boot section.
-    attr_reader :default
-
     def initialize
       @all = []
       @default = ""
     end
 
+    # @return [String] title of default boot section.
+    def default
+      return unless @data
+
+      entry = @data.select { |d| d["id"] == @default }
+      if entry.empty?
+        ""
+      else
+        entry.first["title"]
+      end
+    end
+
     # Sets default section internally.
     # @param [String] value of new boot title to boot
     # @note to write it to system use #write later
     def default=(value)
-      log.info "set new default to '#{value.inspect}'"
-
-      # empty value mean no default specified
-      if !all.empty? && !all.include?(value) && !value.empty?
-        log.warn "Invalid value #{value} trying to set as default. Fallback to default"
-        value = ""
+      entry = @data.select { |d| d["title"] == value }
+      if entry.empty?
```

Comment:
same here. The used method should not be select, but find that return first entry or nil.